src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -398,11 +398,12 @@ public virtual void Write(ReadOnlySpan<byte> buffer)
             }
             else
             {
-                byte[] array = ArrayPool<byte>.Shared.Rent(buffer.Length);
+                int length = buffer.Length;
+                byte[] array = ArrayPool<byte>.Shared.Rent(length);
```

Comment:
This sensitivity to the exact callsite pattern is pretty unfortunate.

Reply:
Issue for it: https://github.com/dotnet/coreclr/issues/15783